flocker/acceptance/test_api.py

```diff
@@ -39,73 +40,314 @@ def api_available():
-            result.addCallback(got_body)
-            return result
-        d.addCallback(lambda _: loop_until(created))
-        return d
+            request.addCallback(got_body)
+            return request
+
+        waiting = loop_until(created)
+        waiting.addCallback(lambda ignored: (self, dataset_properties))
+        return waiting
+
+    def create_dataset(self, dataset_properties):
+        """
+        Create a dataset with the supplied ``dataset_properties``.
+
+        :param dict dataset_properties: The properties of the dataset to
+            create.
+        :returns: A 2-tuple of (cluster, api_response)
+        """
+        request = post(
+            self.base_url + b"/configuration/datasets",
+            data=dumps(dataset_properties),
+            headers={b"content-type": b"application/json"},
+            persistent=False
+        )
+
+        request.addCallback(content)
+        request.addCallback(loads)
+        # Return cluster and API response
+        request.addCallback(lambda response: (self, response))
+        return request
+
+    def update_dataset(self, dataset_id, dataset_properties):
+        """
+        Update a dataset with the supplied ``dataset_properties``.
+
+        :param bytes dataset_id: The uuid of the dataset to be modified.
+        :param dict dataset_properties: The properties of the dataset to
+            create.
+        :returns: A 2-tuple of (cluster, api_response)
+        """
+        request = post(
+            self.base_url + b"/configuration/datasets/%s" % (
+                dataset_id.encode('ascii'),
+            ),
+            data=dumps(dataset_properties),
+            headers={b"content-type": b"application/json"},
+            persistent=False
+        )
+
+        request.addCallback(content)
+        request.addCallback(loads)
+        # Return cluster and API response
+        request.addCallback(lambda response: (self, response))
+        return request
+
+
+def cluster_for_test(test_case, node_addresses):
+    """
+    Build a ``Cluster`` instance with the supplied ``node_addresses``.
+
+    ``flocker-control`` is started on the node with the lowest address and with
+    a blank database.
+
+    ``flocker-zfs-agent`` is started on every node.
+
+    The processes will be killed after each test.
+
+    :param TestCase test_case: The test case instance on which to register
+        cleanup operations.
+    :param list node_address: The IPv4 addresses of the nodes in the cluster.
+    :returns: A ``Cluster`` instance.
+    """
+    # Start servers; eventually we will have these already running on
+    # nodes, but for now needs to be done manually.
+    # https://clusterhq.atlassian.net/browse/FLOC-1383
+
+    control_service = remote_service_for_test(
+        test_case,
+        sorted(node_addresses)[0],
+        [b"flocker-control",
+         b"--data-path",
+         b"/var/lib/flocker/%s" % (random_name(),)]
```

Comment:
We mostly test against throw-away machines, but still, nicer to use `/tmp` if it's test-created garbage.